lib/ansible/module_utils/facts.py

```diff
@@ -3169,8 +3169,44 @@ def populate(self):
         return self.facts
 
     def get_virtual_facts(self):
-        self.facts['virtualization_type'] = ''
-        self.facts['virtualization_role'] = ''
+        sysctl_path = self.module.get_bin_path('sysctl', required=True)
```

Comment:
It's on my list to fix those, not asking you to do that, just this case so i have less work later (im selfish like that!).

I would say an empty string would be less error prone for uses so they don't have to check if the variable is defined.